code/datums/components/itemblock/saberblock.dm

```diff
@@ -46,7 +46,11 @@ TYPEINFO(/datum/component/itemblock/saberblock)
 //proc that is called when the base item is used to block. The parent itemblock component has already registered this proc for the "COMSIG_ITEM_BLOCK_BEGIN" signal
 /datum/component/itemblock/saberblock/on_block_begin(obj/item/I, var/obj/item/grab/block/B)
 	. = ..()//Always call your parents
-	if(!can_block_check || isnum(can_block_check) || (call(I, can_block_check)()))
+	if (istype(I, /obj/item/toy/sword))
+		RegisterSignal(B.assailant, COMSIG_ATOM_PROJECTILE_REFLECTED, PROC_REF(do_reflect_animation))
+		B.setProperty("toyreflection", 1)
+
+	else if(!can_block_check || isnum(can_block_check) || (call(I, can_block_check)()))
```

Comment:
this handling isn't really ideal - would rather have dart-reflection-blocking be its own component, rather than have the saberblock component be typechecking like this

Reply:
i will make my attempts